uvtrick/__init__.py

```diff
@@ -83,6 +81,54 @@ def __init__(self, *requirements: str, python: str = None, debug: bool = False):
         self.requirements = requirements
         self.python = python
         self.debug = debug
+        self.temp_dir: Path = None
+
+    @property
+    def inputs(self) -> Path:
+        return self.temp_dir / "pickled_inputs.pickle"
+
+    @property
+    def script(self) -> Path:
+        return self.temp_dir / "pytemp.py"
+
+    @property
+    def output(self) -> Path:
+        return self.temp_dir / "tmp.pickle"
+
+    @property
+    def cmd(self) -> list[str]:
+        quiet = [] if self.debug else ["--quiet"]
+        deps = [f"--with={dep}" for dep in self.requirements]
+        pyversion = [f"--python={self.python}"] if self.python else []
+        return ["uv", "run", *quiet, *deps, *pyversion, str(self.script)]
+
+    def report(self, contents: str) -> None:
+        """Log the temporary dir, input kw/args and intermediate script to STDOUT."""
+        print(f"Running files in {self.temp_dir}\n{self.cmd}")
+        args, kwargs = pickle.loads(self.inputs)
+        print(f"Pickled args: {args}")
+        print(f"Pickled kwargs: {kwargs}")
+        print(f"Contents of the script:\n\n{contents}")
+        return
+
+    def maincall(self, func) -> str:
+        """A main block to deserialise a function signature then serialise a result.
+
+        Load the args/kwargs from an 'inputs' pickle, call a Python function
+        with them, and store the result in an 'output' pickle.
+        """
+        func_name = func.__name__
+        inputs_path, output_path = self.inputs, self.output
+        return textwrap.dedent(f"""
+        if __name__ == "__main__":
+            import pickle
+            from pathlib import Path
+
+            args, kwargs = pickle.loads(Path('{inputs_path!s}').read_bytes())
+            result = {func_name}(*args, **kwargs)
+            Path('{output_path!s}').write_bytes(pickle.dumps(result))
+        """)
+
 
     @staticmethod
```

Comment:
I noticed a test fail after merging but this is also making me wonder ... should this still be a static method? Might be best as a normal method because the input/output paths are now stored on self.

Reply:
Ah, no it shouldn't be static, I was just trying to make the refactor steps clear in PRs. Was not meant to end up as static, eek

Reply:
Lol, the merge logic has decided to put both forms of the `maincall` method in :joy:

Reply:
Specifically it put the new regular method before the old staticmethod so the new one got overwritten hence tests failed. Patched :adhesive_bandage: